TextToSpeech/impl/TTSSpeaker.cpp

```diff
@@ -885,53 +984,42 @@ std::string TTSSpeaker::constructURL(TTSConfiguration &config, SpeechData &d) {
         return "";
     }
 
-    // EndPoint URL
-    std::string tts_request;
-    TTSURLConstructer url;
-    tts_request = url.constructURL(m_defaultConfig,d.text,false);
-    TTSLOG_WARNING("Constructured final URL is %s", tts_request.c_str());
+    TTSURLConstructer urlConstructor;
+    std::string tts_request = urlConstructor.constructURL(m_defaultConfig, d.text, false, shouldUseLocalEndpoint());
+    if(m_defaultConfig.hasValidLocalEndpoint()) {
+       PipelineType pipelineType = getUrlPipelineType(tts_request);
+       if(pipelineType != m_pipelinetype) {
+          //pipeline switch required
+          TTSLOG_INFO("pipeline needs updation");
+          destroyPipeline();
+          createPipeline(pipelineType);
+       } else {
+          TTSLOG_INFO("re-use existing pipeline.");
+       }
+    }
+    TTSLOG_WARNING("Constructed final URL is %s", tts_request.c_str());
     return tts_request;
 }
 
 void TTSSpeaker::speakText(TTSConfiguration config, SpeechData &data) {
     m_isEOS = false;
     m_duration = 0;
 
-    if(m_pipeline && !m_pipelineError && !m_flushed) {
+    if((m_pipeline && !m_pipelineError && !m_flushed) || m_remoteError) {
```

Comment:
Not sure, if the condition is right. We should proceed to speak if the pipeline exists and client hasn't flushed the request. But should analyze why m_pipelineError was checked, was that to ensure pipeline is recreated after a playback failure? If so it should be handled differently than keeping it and then checking exclusively for m_remoteError. What this means is that irrespective of pipeline's presence & flushed state if the previous request failed with remoteError you proceed to speak now.